### PR TITLE
Refactor eval abstractions to be more general

### PR DESCRIPTION
**Pull request by AntonOsika**
Refactors into:

- Agent
- ExecutionEnv
- FileStore (used by both agent and SimpleExecutionEnv in this case)

tests remain the same.

I liked the breakdown a lot.

But I don't like that we would couple the execution env with the agent since it blocks people to use these abstractions if they want to e.g. sandbox the execution env.

I prefer that the agent produces and artifact (Files), that is testable in different execution envs.

### CODE REVIEW

File: eval/main.py [hunk 1]
```diff
@@ -219,47 +260,24 @@ def run(self, command) -> tuple[str, str, int]:
...
-        return files
-
 
 def execute(test: ExecTest) -> TestResult:
     """
     Executes the code.
     """
     print(f"Running test {test['name']} with prompt: {test['prompt']}")
-    env = GPTMeExecEnv()
+    agent = GPTMe()
```
Comment (AntonOsika) [inline eval/main.py:269]: Reviewing myself, nitpick:

I realize we don't need to instantiate an object in this case.

The agent.act() can just be a function.

File: eval/main.py [hunk 2]
```diff
@@ -158,25 +151,59 @@ class ExecTest(TypedDict):
...
+                        content = f.read()
+                    except UnicodeDecodeError:
+                        content = "binary file"
+                    files[str(path.relative_to(self.working_dir))] = content
+        return files
+
+class Agent:
+    def act(files: Files | None, command: str | None, prompt: str) -> Files:
+        """
+        Returns a dict of actions to take.
```
Comment (AntonOsika) [inline eval/main.py]:
```suggestion
        Carries out the prompt and returns artifacts in the form of `Files`.
```

**Comment by ErikBjare** [comment]
Very nice!

Tests just failing due to no access to the `OPENAI_API_KEY` secret.

Merging.

**Comment by ErikBjare** [comment]
Made some changes fixing the remaining issues: https://github.com/ErikBjare/gptme/commit/cefbbe868fdc2dfd6b3e311e23397b6639c2eb21

**Pull request merged**